apps/extension/src/routes/popup/approval/transaction/index.tsx

```diff
@@ -47,7 +50,18 @@ export const TransactionApproval = () => {
 
   return (
     <div className='flex h-screen flex-col'>
-      <div className='grow overflow-auto p-[30px] pt-10'>
+      <div className='flex grow flex-col overflow-auto p-[30px] pt-10'>
+        {selectedTransactionViewName === TransactionViewTab.SENDER && symbol !== 'UM' && (
```

Comment:
suggestion: think this would be simpler if we ditched the whole useEffect pattern and just did something like:

```ts
const hasAltGasFee = (txv?: TransactionView): boolean => {
  const { stakingAssetId } = new ChainRegistryClient().bundled.globals();
  return txv?.bodyView?.transactionParameters?.fee?.assetId?.equals(stakingAssetId) ?? false;
};
...

  return (
    <div className='flex h-screen flex-col'>
      <div className='flex grow flex-col overflow-auto p-[30px] pt-10'>
        {selectedTransactionViewName === TransactionViewTab.SENDER &&
          hasAltGasFee(selectedTransactionView) && (
            <div
                 ...
```

Reply:
helper function is a better approach here 👍

Reply:
```ts

const hasAltGasFee = (txv?: TransactionView): boolean => {
  const { stakingAssetId } = new ChainRegistryClient().bundled.globals();
  
  let feeAssetId = txv?.bodyView?.transactionParameters?.fee?.assetId;
  if (feeAssetId === undefined) {
    feeAssetId = stakingAssetId;
  }

  return stakingAssetId.equals(stakingAssetId) ?? false;
};
```
...

```ts
{selectedTransactionViewName === TransactionViewTab.SENDER && !hasAltGasFee(selectedTransactionView) && (
```